main.go

```diff
@@ -80,8 +82,15 @@ func main() {
 		cli.WithCLIVersion(buildVersion, buildTime),
 		cli.WithKubernetesNodeVersion(k8sVersion),
 		cli.WithProvider("systemd", func(cfg provider.InitConfig) (provider.Provider, error) {
-			cfg.ConfigPath = o.KubeConfigPath
-			p, err := systemd.New(ctx, cfg)
+			initCfg := systemd.InitConfig{InitConfig: cfg}
+			initCfg.ConfigPath = o.KubeConfigPath
+			initCfg.SystemdUser = user
+			initCfg.UnitDir = "/var/run/systemk"
```

Comment:
I think we have to fail early and not let systemk even talk to Kubernetes or systemd before we ensure the specified user can actually write in this directory.

Reply:
we don't write there, systemd does that. Currently we fail if we can't create our `systemk` subdir